charts/openshift-metering/templates/hive/hive-scripts-configmap.yaml

```diff
@@ -109,4 +109,35 @@ data:
     export HIVE_METASTORE_HADOOP_OPTS=" -Dhive.log.level=${HIVE_LOGLEVEL} "
     export HIVE_OPTS="${HIVE_OPTS} --hiveconf hive.root.logger=${HIVE_LOGLEVEL},console "
 
+    # Note(tflannag): Hive supports derby (default), mysql, postgres, and oracle databases,
+    # but we only allow users to configure a mysql or postgres database instance.
+    #
+    # The following attempts to determine the database that was configured by the metering
+    # operator by parsing the ConnectionURL parameter in hive-site.xml. We can make the same
+    # assumption that this file exists and that parameter has been set as we control that
+    # file as part of Hive's reconciliation process. We use faq to pull out that key's
+    # value and parse the URL's scheme. We can then use the value of that scheme in order
+    # to avoid determing what database needs to be passed to `schematool -dbType`.
+    #
+    # TODO(tflannag): we may want to use a more dedicated tool in order to parse a URL
+    # as there's currently no validation in place such that invalid ConnectionURL values
+    # being configured may result in unexpected values being returned. Going forward, we
+    # should likely use Ansible to parse this URL for us and setting that environment variable
+    # to avoid determining it using bash.
+    export SCHEMATOOL_BIN="${HIVE_HOME}/bin/schematool"
+    DATABASE_TYPE=$(faq -f xml -o json \
+        '.configuration.property[] | select (.name=="javax.jdo.option.ConnectionURL").value' \
+        "${HIVE_HOME}/conf/hive-site.xml" | \
+        awk -F [/:] '{ print $2 }')
+
+    # Check if there's existing metastore schema for that ${DATABASE_TYPE} and if no
+    # schema exists, i.e. no errors was returned when using the `-info` parameter, attempt
+    # to initialize that schema. Note: running schematool when using derby, which is what
+    # metering deploys by default, can be problematic during upgrades so only run it when
+    # mysql is the underlying database.
+    echo "Deploying Hive using ${DATABASE_TYPE} as the underlying database"
+    if [[ $DATABASE_TYPE == "mysql" ]]; then
+        "${SCHEMATOOL_BIN}" -dbType ${DATABASE_TYPE} -info || "${SCHEMATOOL_BIN}" -dbType ${DATABASE_TYPE} -initSchema
+    fi
```

Comment:
I was seeing failures in the post-upgrade CI job that deploys derby in a previously released version, or metering then upgrades to using an image that contains these hive-scripts ConfigMap fixes. We haven't seen any problems with derby in the past, and it looks like only mysql (haven't tested postgres yet) require running the `schematool` CLI to get Hive and MySQL playing nicely.